plugins/module_utils/ansible_freeipa_module.py

```diff
@@ -220,25 +240,35 @@ def compare_args_ipa(module, args, ipa):
             if isinstance(ipa_arg, list):
                 if not isinstance(arg, list):
                     arg = [arg]
+                if len(ipa_arg) != len(arg):
+                    module.debug(
+                        base_debug_msg
+                        + "List length doesn't match for key %s: %d %d"
+                        % (key, len(arg), len(ipa_arg),)
+                    )
+                    return False
                 if isinstance(ipa_arg[0], str) and isinstance(arg[0], int):
                     arg = [to_text(_arg) for _arg in arg]
                 if isinstance(ipa_arg[0], unicode) and isinstance(arg[0], int):
                     arg = [to_text(_arg) for _arg in arg]
-            # module.warn("%s <=> %s" % (repr(arg), repr(ipa_arg)))
             try:
                 arg_set = set(arg)
                 ipa_arg_set = set(ipa_arg)
             except TypeError:
                 if arg != ipa_arg:
-                    # module.warn("%s != %s" % (repr(arg), repr(ipa_arg)))
+                    module.debug(
+                        base_debug_msg
+                        + "Different values: %s %s" % (arg, ipa_arg)
+                    )
```

Comment:
Is this a debug message left behind or is it intentional? (I kinda like it.)

Reply:
Yes it was left intentionally.

That brings something I'd like to ask you and @t-woerner about. We are forcing debug mode in all modules (or at least the modules I've checked). Is that really needed? Why did we decide to do that over only enabling debug when it's needed.

Reply:
I'm not sure why debug mode is always enable.

Reply:
What about, removing that in the class? It wouldn't affect the modules not using the class.

Reply:
I'm not sure. But adding it later is easy, and we can add only if needed. Please, remove it.

Reply:
Done.